ViAn/GUI/projectwidget.cpp

```diff
@@ -614,28 +625,61 @@ void ProjectWidget::remove_item() {
     QString info_text = "Do you wish to continue?";
     if (message_box(text, info_text, true)) {
         for (auto item : selectedItems()) {
-            if (item->type() == FOLDER_ITEM) {
-                std::vector<VideoItem*> v_items;
-                get_video_items(item, v_items);
-                for (auto v_item : v_items) {
-                    emit item_removed(v_item->get_video_project());
-                }
-            }
-
-            else if (item->type() == VIDEO_ITEM) {
-                VideoItem* vid_item = dynamic_cast<VideoItem*>(item);
-                emit item_removed(vid_item->get_video_project());
-            }
-            // TODO Fix these cases
-//            else if (item->type() == TAG_ITEM || item->type() == ANALYSIS_ITEM) {
-//
-//            }
-            delete item;
+            remove_tree_item(item);
         }
     }
     emit remove_overlay();
 }
 
+/**
+ * @brief ProjectWidget::remove_tree_item
+ * @param item
+ * Recursive function for removing item and all of its children
+ */
+void ProjectWidget::remove_tree_item(QTreeWidgetItem* item) {
+    if (item->type() == FOLDER_ITEM) {
+        while (item->childCount() != 0) {
+            remove_tree_item(item->child(0));
+        }
+    }
+    else if (item->type() == VIDEO_ITEM) {
+        VideoItem* vid_item = dynamic_cast<VideoItem*>(item);
+        VideoProject* v_proj = vid_item->get_video_project();
+
+        // Remove all children
+        while (item->childCount() != 0) {
+            remove_tree_item(item->child(0));
+        }
+        // Remove the video from the list of videos
+        auto it = std::find(m_proj->get_videos().begin(), m_proj->get_videos().end(), v_proj);
```

Comment:
Is the video project ever deleted? Else it needs to be. Same goes for all removed items, their underlying data structure needs to be properly deleted.